### PR TITLE
Add option to remove empty subdirectories

### LINKED ISSUE #11: Add flag to delete empty directories
Housekeeper should have an optional flag to remove empty subdirectories.

`housekeeper.exe --ext "log" --older-than 30 --path "c:\logs" --recursive --case-insensitive --remove-directories` 

should remove any subdirectories that are now empty, but leave c:\logs intact.

### PR DESCRIPTION
**Pull request by rokett**
Closes #11

Optional `--remove-directories` flag to remove any empty subdirectories when doing a recursive search.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,21 +6,22 @@ Small app to delete specified files older than the specified number days from a
...
+| debug              | Enable debug logging.                                                    | false         |
+| version            | Display version and build number                                         | false         |
+| case-insensitive   | Match file extensions regardless of case                                 | false         |
+| remove-directories | Remove **empty** subdirectories when doing a recursive search            | false         |
+
+For example; delete files with the **.log** file extension, ignoring case, which are older than **30** days and are anywhere within the **c:\logs** directory.  Additionally remove any empty directories within the **c:\logs** directory.
 
 ````Batchfile
-housekeeper.exe --ext "log" --older-than 30 --path "c:\logs" --recursive --case-insensitive
+housekeeper.exe --ext "log" --older-than 30 --path "c:\logs" --recursive --case-insensitivem --remove-directories
```
Comment (sparkblaze) [inline README.md]:
```suggestion
housekeeper.exe --ext "log" --older-than 30 --path "c:\logs" --recursive --case-insensitive --remove-directories
```

**Comment by sparkblaze** [comment]
When the last folder in the path flag has a space in ( `--path "c:\temp\New Folder` ), it throws an error 

> The filename, directory name, or volume label syntax is incorrect.

Some directories are left behind - possibly because it isn't going back up the directory tree?

using my logstash config directory, it's leaving the master branch folder and the conf.d folder intact - despite being empty. Running the command again removes the conf.d directory, a 3rd time will remove the master branch directory.

**Comment by rokett** [comment]
oddly I don't get the same error when testing with a root folder that has a space in it.  When I tried it, it worked as expected.

To clarify, your root folder is `c:\temp\new folder`?  You have other folders and files underneath that one?

**Comment by sparkblaze** [comment]
My bad, looks like it's the trailing slash that it doesn't like :)

We should still resolve that if we can though :)

**Comment by rokett** [comment]
ahhh there's a trailing slash.....that's easy enough as I can just strip it from the root path.  I'll test that.

**Comment by rokett** [comment]
actually even with a trailing slash I don't get that error....it actually prints the help text and then exits.  I can't seem to replicate your issue at the moment.

**Comment by sparkblaze** [comment]
This is what I have running `.\Housekeeper_1.2.0.windows.amd64.exe --ext * --older-than 1 --path 'C:\temp\New folder\' --recursive --case-insensitive --remove-directories`

**Comment by rokett** [comment]
huh.....I can replicate your issue with your command, but I can't see what's different to the one I typed in manually other than the order of the flags, which really shouldn't matter.

It looks like it's interpreting the rest of the arguments as part of the path, but I'm not clear why.  I'll need to look into that bug a bit more.

**Comment by rokett** [comment]
oh actually I do know why.....the trailing slash escapes the final quote so it really does think the rest of the command is part of the path arguments.

Unfortunately it does that prior to me being able to affect the flags, so I don't think stripping out the trailing slash will help.  I'll have a think about how to deal with it, but it may be a case of adding an important note to the readme to not use trailing slashes.

**Comment by rokett** [comment]
I can't think of a way to deal with the trailing slashes in the path; there's not much I can do in-app I believe as it's just how Windows works.  I've added some explanatory text to the readme for now.

I **think** it will also now recursively delete sub directories as you expect; give it a go.

**Pull request merged**